spec/porky_lib/file_service_spec.rb

```diff
@@ -94,227 +95,235 @@ def write_test_file(data)
     tempfile
   end
 
-  it 'write encrypted data to S3' do
-    file_key = file_service.write(plaintext_data, bucket_name, default_key_id)
-    expect(file_key).not_to be_nil
-  end
-
-  it 'write large encrypted data to S3' do
-    file_key = file_service.write(File.read("spec#{File::SEPARATOR}porky_lib#{File::SEPARATOR}data#{File::SEPARATOR}large_plaintext"),
-                                  bucket_name, default_key_id)
-    expect(file_key).not_to be_nil
-  end
+  describe '#write' do
```

Comment:
I don't understand why all these changes to #write tests? Was this impacted by your change.
We should also compose better descriptions for the tests, what you have it will end up like: 
#write write encrypted data to S3